gprofiler/main.py

```diff
@@ -306,6 +313,16 @@ def _snapshot(self):
             else:
                 logger.info("Successfully uploaded profiling data to the server")
 
+    def _log_kernel_message_oom(self, message):
+        entry = get_oom_entry(message)
+        if entry:
+            logger.info(f"OOM: {json.dumps(entry._asdict())}")
```

Comment:
We want 2 modes for OOMs & segfaults:

* Standard: avoid using AP if a process that we've profiled dies (so we only log it, and care about it, if the PID has been profiled by us).
* Safemode (`--java-safemode`) - more paranoid, logs & reacts upon every segfault/OOM.

Reacting to OOMs - we'll add that later (that is, disabling the use of AP).
Logging - we can add it now - different log message if the dying process was profiled by us (I suppose that `JavaProfiler` could keep a list of the PIDs it has profiled in its last interval). And if we're not in `--java-safemode` (the paranoid one) - do not log all messages as we don't care about them

Reply:
now logging only profiled pids